SteamProject/MySteam/Steam.cpp

```diff
@@ -7,6 +7,7 @@ void FSteam::RunApp()
 {
 	while (true)
 	{
+		system("cls");
 		int UserOption;
 		UserOption = ValidateInput("------ Welcome to Steam! ------\n\t1 - Add New game \n\t2 - Categories Menu... \n\t3 - Show Games... \n\t4 - Exit \n");
```

Comment:
Passing the question to repeat like this looks a bit cumbersome, as its passing "an entire menu" by parameter. I would recommend printing it to console outside this function, before calling it, as it's printing only once on the function anyways. You can also tidy this up even more by using `std::endl` so you can print out the menu line by line, so that the code resembles the output, improving readability of the code. For example:

```
std::cout << "Welcome to Steam!" << std::endl;
std::cout << "1 - Add New Game" << std::endl;
```

Reply:
I completely agree with you @GaxysGalactic. I didn't really catch this in the previous review, but you shouldn't be passing the whole contents of your menu into a function named `ValidateInput`.

However, it makes sense in your design, since you want to clear the screen and print the whole message again. In that case, I'd say it's fine but the name of the function is the one that's incorrect. 

`ValidateInput` seems like you're validating some input, but you're not even passing in anything to validate. 

In that case, what about something like `GetInputForQuestion` or `GetValidInputForQuestion`. But keep in mind that you are not validating if the input is within the correct range, so if you're supposed to get a valid input, you may want to do that.

Reply:
Regarding the std::endl thing, you could do a couple things:
1. You could use an `std::stringstream` which let's you construct a string in a similar syntax as std::cout. Something like:
```
std::stringstream QuestionStream;
QuestionStream<< "Welcome to Steam!" << std::endl;
QuestionStream<< "1 - Add New Game" << std::endl;
// blah blah
const int UserOption = GetValidInputForQuestion(Stream.str());
```
2. You could use raw string literals, which allows you to create multiline strings without the need of \n or endl:
```
std::string Question = R"(
Welcome to Steam!
1 - Add New Game
Blah Blah
");
const int UserOption = GetValidInputForQuestion(Question);
```